### DISCUSSION TITLE
Debug embedded python module in c++

### DISCUSSION THREAD

**Discussion by abhishekmonu96** [pybind/pybind11#4545] [Q&A]
I have a C++ code in which I am calling a python module using pybind11. Can any one tell me how to debug the python part of the code. The python module is custom. May be by debugger attachment or any other method that can help me?

**Comment by Dariusz1989** [comment]
u want to debug pyd or python code?
If python, u can attach debugger, pycharm pro has remote debugger u can use to attach inside ur code
U just do
import sys
sys.path.append(pathToDebugger/pycharmpro/libs/)
debugger.start() // < I dont remember exact code)
then u shpuld be able to debug it.
if u want pyd debugging, then u need visual studio code, attach remote process and break point.
Make sure u build in debug/releaseWithDebInfo and u can breakpoint. Be sure u have symbols loaded.

**Reply by abhishekmonu96** [reply]
Thank you for the response! This will help me a lot. I will get back after trying it. Hope this resolves my problem!

**Reply by Dariusz1989** [reply]
Sorry, visual studio, not code!! That VS can attach at runtime to process & breakpoint inside pyd compiles source C++.

**Comment by zeroxer** [comment]
I use `PYBIND11_EMBEDDED_MODULE`. Is there any mothod to step debug python code in vscode?

**Reply by Dariusz1989** [reply]
No, you need visual studio. pyd is c++ dll extension.
